wallet/lnd/lnd.go

```diff
@@ -53,71 +51,83 @@ func NewLNDWallet(macaroonPath string,  address string) *LNDWallet {
 	}
 }
 
-// PayInvoice attempts to pay the given invoice and returns the result.
-func (lndw *NewLNDWallet) PayLndInvoice(ctx context.Context, invoice wallet.Invoice) (*wallet.PaymentLndResult, error) {
-	path := "/v2/invoices/settle"
-	body := map[string]interface{}{
-		"preimage": invoice,
-	}
 
-	responseBody, err := lndw.makeRequest(ctx, "POST", path, body)
+func (lndw *NewLNDWallet) HandleSettleInvoice(ctx context.Context, invoice wallet.Invoice) (*wallet.PaymentLndResult, error) {
```

Comment:
Please adhere to already existing Wallet interface as per Issue description and requirements. As it stands this wallet does not implement the Wallet interface.